### ISSUE TITLE
RefBy fields can contain references to themselves

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#1210]
If an IM references itself, it will show up in the RefBy field for that IM, which seems unnecessary. The first IM in SSP is an example:
![image](https://user-images.githubusercontent.com/18634568/56317087-fc8a8280-6129-11e9-9472-040b921375c5.png)

(The self-reference isn't visible in the screenshot, it happens in the derivation)

**Comment by smiths** [comment]
@bmaclach, I agree that the self reference should not appear in the RefBy field.  I had a look at this particular reference of this IM to itself.  If we had to, we could delete it.  The purpose of the reference is for the convenience of the reader, especially since the derivation is (necessarily) so long.  I would like to keep the ability for a IM to reference itself, for the purposes of clear documentation.

**Comment by JacquesCarette** [comment]
We should be able to filter out self-references easily enough. I'm going to have some real coding to do this summer!

**Comment by balacij** [comment]
I'm marking this as 'blocked' (on #2873) because the path to finding when things reference themselves is unclear right now. It might be best to do this ticket _after_ #2873 is resolved because a part of #2873's proposed solution is that each chunk contains a list of all chunks it references (depends) on (by `UID`).
